### PR TITLE
Fix backward propagation

### PR DESCRIPTION
**Pull request by LuEdRaMo**
Currently, backward propagation throws an error because this line:
https://github.com/PerezHz/NEOs.jl/blob/905a80e0c40640db0df0904726296d70c1ec2c0d/src/propagation/propagation.jl#LL324C1-L324C60
loads an incomplete time interval. This PR introduces a simple change to fix this.

### CODE REVIEW

File: src/observations/jpl_eph.jl [hunk 1]
```diff
@@ -124,20 +124,28 @@ const ttmtdb = TaylorInterpolant(sseph.t0, sseph.t, sseph.x[:,end])
...
 function loadpeeph(et::Union{Nothing, Real} = nothing)
     if isnothing(et)
         return sseph
     else
-        idxs = findall(x -> x <= et, sseph.t)
-        return TaylorInterpolant(sseph.t0, sseph.t[idxs], sseph.x[idxs[1:end-1], :])
+        i = searchsortedfirst(sseph.t, et)
+        return TaylorInterpolant(sseph.t0, sseph.t[1:i], sseph.x[1:i-1, :])
     end
 end
```
Comment (lbenet) [inline src/observations/jpl_eph.jl:141]: Two comments (about this function):
- I think it makes sense to exploit dispatch here, and define the functions
    ```julia
    loadpeeph() = sseph
    loadpeeph(::Nothing) = loadpeeph()
    ```
    In this way, you avoid the the `Union`, the `if` and thus all branching.
- What is `sseph` here and in my previous suggestion? My point is that it is either a `global` quantity, which we should perhaps avoid, or simply undefined...

Comment (PerezHz) [reply src/observations/jpl_eph.jl:141]: > I think it makes sense to exploit dispatch here, and define the functions

+ 1 to the suggestion of using multiple dispatch 👍

**Comment by PerezHz** [comment]
> Currently, backward propagation throws an error because this line:

Woops, sorry about that! I suggest to add also backward propagation to the tests!

**Comment by PerezHz** [comment]
And many thanks for the fix @LuEdRaMo!

File: src/observations/jpl_eph.jl (hunk 1 above)
Comment (PerezHz) [reply src/observations/jpl_eph.jl:141]: Regarding `sseph`, I suggest to type-annotate it, to help the compiler with type-inference:
```julia
const sseph::TaylorInterpolant{Float64,Float64,2} = JLD2.load(sseph_artifact_path, "ss16ast_eph")
```
We could even consider adding type annotations to `loadpeeph`:
```julia
loadpeeph()::TaylorInterpolant{Float64,Float64,2} = sseph
loadpeeph(::Nothing)::TaylorInterpolant{Float64,Float64,2} = loadpeeph()
```
The rationale behind using `sseph` is to load the the Solar System ephemeris  via artifacts only once in memory.

**Comment by PerezHz** [comment]
I went ahead and included a couple of unit tests for backward propagations, and bumped the patch version. I propose to merge this and address further issues (e.g., #19) in separate PRs. Again, great catch and many thanks for this fix, @LuEdRaMo!

**Pull request merged**